pkg/activator/net/revision_backends.go

```diff
@@ -397,9 +397,11 @@ func (rw *revisionWatcher) checkDests(curDests, prevDests dests) {
 	if ok, err := rw.probeClusterIP(dest); err != nil {
 		rw.logger.Errorw("Failed to probe clusterIP "+dest, zap.Error(err))
 	} else if ok {
-		// We can reach here only iff pods are not successfully individually probed
+		// We can reach here only if pods are not successfully individually probed
 		// but ClusterIP conversely has been successfully probed.
-		rw.podsAddressable = false
+		if rw.meshMode == netcfg.MeshCompatibilityModeEnabled {
```

Comment:
@ReToCode thoughts on this one?

Reply:
Hm, I'm not sure about all the implications here. Last time I did dig down that rabbithole, this flag had more effects than just that.

Also, instead of setting this, probably Activator should rather watch that config and reconfigure instead of needing the restart.